SplunkAppForWazuh/appserver/static/js/controllers/overview/aws/awsCtrl.js

```diff
@@ -2,74 +2,156 @@ define([
   '../../module',
   '../../../services/visualizations/chart/pie-chart',
   '../../../services/visualizations/chart/area-chart',
+  '../../../services/visualizations/chart/column-chart',
   '../../../services/visualizations/table/table',
   '../../../services/visualizations/map/map',
-  '../../../services/visualizations/inputs/time-picker'
-], function(app, PieChart, AreaChart, Table, Map, TimePicker) {
+  '../../../services/visualizations/inputs/time-picker',
+  '../../../services/visualizations/inputs/dropdown-input'
+], function (app, PieChart, AreaChart, ColumnChart, Table, Map, TimePicker, Dropdown) {
   'use strict'
 
   class AWS {
-    constructor($urlTokenModel, $scope, $currentDataService, $state) {
+    constructor($urlTokenModel, $rootScope, $scope, $currentDataService, $state, awsMetrics) {
+      this.rootScope = $rootScope
       this.scope = $scope
       this.state = $state
+      this.awsMetrics = awsMetrics
       this.currentDataService = $currentDataService
       this.currentDataService.addFilter(`{"rule.groups":"amazon", "implicit":true}`)
       this.getFilters = this.currentDataService.getSerializedFilters
       this.filters = this.getFilters()
+      this.amazonFilters = this.buildAmazonFilter()
+      this.implicitFilters = this.serializedImplicitFilters(this.currentDataService.getFilters().filter(item => item['implicit']))

```

Comment:
If `getFilters` fails, this throws an exception. Don't do unsafe things in a `constructor`